### DISCUSSION TITLE
SDK - Get Item ID after createOne?

### DISCUSSION THREAD

**Discussion by daltonsutton** [directus/directus#12819] [Archive]
I'm having a hard time finding a way to get the ID of the item after using createOne. I've looked through the documentation and couldn't find any examples.

I have tried something like the following without any success:

```js
const post = posts.createOne({
    content: row.content,
    location: row.location,
    tags: row.tags,
    link: row.link,
    status: row.status,
    date_created: row.date_created,
    user_created: row.user_created,
});

const post_id = post.id;
```

Can someone point me in the right direction? Thanks!

**Comment by rijkvanzanten** [comment]
Does the user you're using for this request have read access to the `posts` collection?

**Reply by daltonsutton** [reply]
I believe so, I'm using an admin account for the authentication.

**Reply by rijkvanzanten** [reply]
Ah in the above bit of code, `createOne` is asynchronous. Have you tried something like:

```js
const post = await posts.createOne({
  ...
});

const post_id = post.id;
```

**Reply by daltonsutton** [reply]
I tried that, but I guess I had something else in the script that was an issue originally. That ended up working great. Thanks for your killer work on Directus and your time and help!
